tests/e2e_test.go

```diff
@@ -0,0 +1,419 @@
+		blockchainIDA, blockchainIDB   ids.ID
+		chainANodeURIs, chainBNodeURIs []string
+		fundedKey                      *ecdsa.PrivateKey
+		fundedAddress                  common.Address
+		teleporterKey                  *ecdsa.PrivateKey
+		err                            error
+		receivedWarpMessage            *avalancheWarp.Message
+		chainAWSClient, chainBWSClient ethclient.Client
+		chainAIDInt                    *big.Int
+		payload                        []byte
+	)
+
+	fundedKey, err = crypto.HexToECDSA(fundedKeyStr)
+	if err != nil {
+		panic(err)
+	}
+	fundedAddress = crypto.PubkeyToAddress(fundedKey.PublicKey)
+
+	teleporterKey, err = crypto.HexToECDSA(teleporterKeyStr)
+	if err != nil {
+		panic(err)
+	}
+
+	ginkgo.It("Setup subnet URIs", ginkgo.Label("Relayer", "Setup"), func() {
+		subnetIDs = manager.GetSubnets()
+		gomega.Expect(len(subnetIDs)).Should(gomega.Equal(2))
+
+		subnetA = subnetIDs[0]
+		subnetADetails, ok := manager.GetSubnet(subnetA)
+		gomega.Expect(ok).Should(gomega.BeTrue())
+		gomega.Expect(len(subnetADetails.ValidatorURIs)).Should(gomega.Equal(5))
+		blockchainIDA = subnetADetails.BlockchainID
+		chainANodeURIs = append(chainANodeURIs, subnetADetails.ValidatorURIs...)
+
+		subnetB = subnetIDs[1]
+		subnetBDetails, ok := manager.GetSubnet(subnetB)
+		gomega.Expect(ok).Should(gomega.BeTrue())
+		gomega.Expect(len(subnetBDetails.ValidatorURIs)).Should(gomega.Equal(5))
+		blockchainIDB = subnetBDetails.BlockchainID
+		chainBNodeURIs = append(chainBNodeURIs, subnetBDetails.ValidatorURIs...)
+
+		log.Info("Created URIs for both subnets", "ChainAURIs", chainANodeURIs, "ChainBURIs", chainBNodeURIs, "blockchainIDA", blockchainIDA, "blockchainIDB", blockchainIDB)
+
+		chainAWSURI := httpToWebsocketURI(chainANodeURIs[0], blockchainIDA.String())
+		log.Info("Creating ethclient for blockchainA", "wsURI", chainAWSURI)
+		chainAWSClient, err = ethclient.Dial(chainAWSURI)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		chainAIDInt, err = chainAWSClient.ChainID(context.Background())
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		chainBWSURI := httpToWebsocketURI(chainBNodeURIs[0], blockchainIDB.String())
+		log.Info("Creating ethclient for blockchainB", "wsURI", chainBWSURI)
+		chainBWSClient, err = ethclient.Dial(chainBWSURI)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		log.Info("Finished setting up e2e test subnet variables")
+	})
+
+	ginkgo.It("Set up relayer config", ginkgo.Label("Relayer", "Setup Relayer"), func() {
+		hostA, portA, err := getURIHostAndPort(chainANodeURIs[0])
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		hostB, portB, err := getURIHostAndPort(chainBNodeURIs[0])
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		log.Info("Setting up relayer config", "hostA", hostA, "portA", portA, "blockChainA", blockchainIDA.String(), "hostB", hostB, "portB", portB, "blockChainB", blockchainIDB.String(), "subnetA", subnetA.String(), "subnetB", subnetB.String())
+
+		relayerConfig := config.Config{
```

Comment:
It might be useful for authoring additional tests to have a global `defaultRelayerConfig` and modify it for individual test cases.